idb/postgres/postgres.go

```diff
@@ -1251,30 +1286,28 @@ ON CONFLICT (addr, assetid) DO UPDATE SET amount = account_asset.amount + EXCLUD
 	if !any {
 		db.log.Debugf("empty round %d", round)
 	}
-	var importState importState
-	staterow := tx.QueryRow(`SELECT v FROM metastate WHERE k = 'state'`)
-	var stateJSONStr string
-	err = staterow.Scan(&stateJSONStr)
+
+	importstate, err := db.getImportState(tx)
 	if err != nil {
-		return
+		return err
 	}
-	err = encoding.DecodeJSON([]byte(stateJSONStr), &importState)
-	if err != nil {
-		return
+
+	if importstate.AccountRound == nil {
```

Comment:
If I'm not mistaken, this is where `importState` should be initialized. So we need to allow the nil case on round 0:
```suggestion
	if importstate.AccountRound == nil && round != 0 {
```

Reply:
This would effect the other errors as well, `err.ErrorNotInitialized` is allowed, and `>= round` below needs to account for the first round.

Reply:
The import state is initialized before the fetcher starts. Here it should already be initialized.

Reply:
I see, so this code only works because there are no transactions in round 0. Maybe that's always been the case.

Reply:
Yes, `commitRoundAccounting()` is not called with `round = 0`.